File: examples/demo-js/src/math.js
export function clamp(value, lo, hi) {
  if (value < lo) return lo;
  if (value > hi) return hi;
  return value;
}

export function average(values) {
  if (values.length === 0) return 0;
  let sum = 0;
  for (const v of values) {
    sum += v;
  }
  return sum / values.length;
}

export function isEven(n) {
  return n % 2 === 0;
}

export function countAbove(values, threshold) {
  let count = 0;
  for (const v of values) {
    if (v > threshold) {
      count += 1;
    }
  }
  return count;
}

export function describe(n) {
  if (n < 0) return "negative";
  if (n === 0) return "zero";
  return "positive";
}

File: examples/demo-cpp/src/math.cpp
#include "math.h"

namespace calc {

int clamp(int value, int lo, int hi) {
    if (value < lo) return lo;
    if (value > hi) return hi;
    return value;
}

double average(const double* values, int count) {
    if (count == 0) return 0.0;
    double sum = 0.0;
    for (int i = 0; i < count; ++i) {
        sum += values[i];
    }
    return sum / count;
}

bool isEven(int n) {
    return n % 2 == 0;
}

int countAbove(const double* values, int count, double threshold) {
    int n = 0;
    for (int i = 0; i < count; ++i) {
        if (values[i] > threshold) {
            n += 1;
        }
    }
    return n;
}

const char* describe(int n) {
    if (n == 0) return "zero";
    if (n < 0) return "negative";
    return "positive";
}

}  // namespace calc
